doc/modules/neighbors.rst

```diff
@@ -510,6 +510,67 @@ the model from 0.81 to 0.82.
     classification using nearest centroid with different shrink thresholds.
 
 
+Nearest Neighbors Transformer
+=============================
+
+Many scikit-learn estimators rely on nearest neighbors: Several classifiers and
+regressors such as :class:`KNeighborsClassifier` and
+:class:`KNeighborsRegressor`, but also some clustering methods such as
+:class:`~sklearn.cluster.DBSCAN` and
+:class:`~sklearn.cluster.SpectralClustering`, and some manifold embeddings such
+as :class:`~sklearn.manifold.TSNE` and :class:`~sklearn.manifold.Isomap`.
+
+All these estimators can compute internally the nearest neighbors, but most of
+them also accept precomputed nearest neighbors :term:`sparse graph`,
+as given by :func:`~sklearn.neighbors.kneighbors_graph` and
+:func:`~sklearn.neighbors.radius_neighbors_graph`. With mode
+`mode='connectivity'`, these functions return an binary adjacency sparse graph
+as required for instance in :class:`~sklearn.cluster.SpectralClustering`,
+whereas with `mode='distance'`, they return a distance sparse graph as required
+for instance in :class:`~sklearn.cluster.DBSCAN`. The benefits of precomputation
```

Comment:
There are several benefits to precomputing this distance matrices?

Reply:
I tried to reformulate.